### PR TITLE
Copy inlined slices gem from house seven

### PR DESCRIPTION
**Pull request by sohohouse-devs**

### CODE REVIEW

**Comment by markburns** [comment]
Accidentally opened PR from wrong browser tab. This is me doing this PR

File: app/models/asset.rb [hunk 1]
```diff
@@ -167,35 +167,78 @@ def search_page_cache
...
+    file_dimensions.keys.each do |key|
       next if key == 'original' || key == 'admin'
-      self.file_dimensions.delete key
+      file_dimensions.delete key
     end
   end
 
   private
 
   def rename_file
```
Comment (markburns) [inline app/models/asset.rb:177]: The actually relevant (non rubocop) changes in the PR are in this file

File: spec/models/asset/rename_spec.rb [hunk 2]
```diff
@@ -23,6 +22,28 @@
     it "moves the file on storage" do
       expect(asset.file).to be_exists
     end
+
```
Comment (markburns) [inline spec/models/asset/rename_spec.rb:25]: More non rubocop changes here

File: app/assets/javascripts/slices/lib/plugins.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ window.log = function(){
   log.history = log.history || [];   // store logs to an array for reference
   log.history.push(arguments);
   if(this.console) {
-    console.log( Array.prototype.slice.call(arguments) );
+    //console.log( Array.prototype.slice.call(arguments) );
```
Comment (dan-palmer) [inline app/assets/javascripts/slices/lib/plugins.js]: Should we just bin these `console.log`s?

File: app/models/asset.rb [hunk 4]
```diff
@@ -6,7 +6,7 @@ class Asset
 
   IMAGE_REGEX = /(jpg|jpeg|gif|png)/
   NORMALIZED_EXTENSIONS = { ".jpeg" => ".jpg" }
-
+  EXTENSION_REGEX = /\.#{IMAGE_REGEX}$/
```
Comment (dan-palmer) [inline app/models/asset.rb]: Should we pop this guy in the `FilenameSanitizer` class?

File: app/models/asset.rb [hunk 5]
```diff
@@ -167,35 +167,78 @@ def search_page_cache
...
-    @renamed = true
   end
 
-  def sanitize(filename)
-    file.send(:cleanup_filename, filename).tap do |name|
-      name.slice!(100..-1)
-      name << SecureRandom.hex(2)
-      name << normalized_extension if file && file.original_filename
+  class FilenameSanitizer
+    QUITE_LONG_FILENAME_LENGTH = 100
```
Comment (dan-palmer) [inline app/models/asset.rb:187]: `QUITE_LOLZ_FILENAME_LENGTH`

**Review by dan-palmer — APPROVED** [review]

**Pull request merged**